runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java

```diff
@@ -408,4 +423,53 @@ private void resetTimers()
 		lastOpponent = null;
 		lastInteracting = null;
 	}
+
+	private void reset()
+	{
+		Splitter COMMA_SPLITTER = Splitter.on(Pattern.compile("\\s*,\\s*"));
+
+		// gets the highlighted items from the text box in the config
+		notifiedMessagesList = COMMA_SPLITTER.splitToList(config.getMessageNotification().trim());
+
+		notifiedMessages = CacheBuilder.newBuilder()
+			.maximumSize(512L)
+			.expireAfterAccess(10, TimeUnit.MINUTES)
+			.build(new WildcardMatchLoader(notifiedMessagesList));
+	}
+
+	@Override
+	protected void startUp()
+	{
+		reset();
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		notifiedMessages.invalidateAll();
+		notifiedMessages = null;
+		notifiedMessagesList = null;
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getGroup().equals("idlenotifier"))
+		{
+			reset();
+		}
+	}
+
+	@Subscribe
+	public void onChatMessageReceived(ChatMessage event)
+	{
+		if (config.message() && isNotificationMessage(event.getMessage()))
+		{
+			notifier.notify(client.getLocalPlayer().getName() + " got message: " + event.getMessage());
+		}
+	}
+	public boolean isNotificationMessage(String message)
+	{
+		return TRUE.equals(notifiedMessages.getUnchecked(message));
+	}
```

Comment:
This is correct code, but Java's native Boolean checking is pretty backwards. You can use Apache Commons' `BooleanUtils.isTrue( )` to write this more like an English sentence.

Reply:
This was how it was done in GroundItems, so I wanted to stay consistent. Either way it works.